CMakeLists.txt

```diff
@@ -6,6 +6,20 @@ cmake_policy(SET CMP0057 NEW)
 
 project(opentelemetry-cpp)
 
+# Mark variables as used so cmake doesn't complain about them
+mark_as_advanced(CMAKE_TOOLCHAIN_FILE)
```

Comment:
What's the CMake complain here? User would not be able to see it in CMake GUI if it is marked as advanced.

Reply:
This is done to suppress "unused variable" on non-Windows OS. If you click on `[x] Advanced` in CMake GUI, then you can still show this variable. It'd be in the same class as a number of other important variables, like `CMAKE_CXX_COMPILER`, `CMAKE_C_COMPILER`, `CMAKE_C_FLAGS` ... I borrowed this idea from vcpkg.

Reply:
Thanks, the mark of advanced makes sense then.